lib/middleware/request-handler.ts

```diff
@@ -163,66 +163,76 @@ function onSubscribe(req: Interface.IOurRequest,
                      res: Interface.IOurResponse,
                      next: restify.Next): void
 {
+    // Validate input options
+    var isValid = true;
+    var errMessage: string;
     var subscriptions: Subscription[] = [];
-    ((): Promise<void> => {
-        // Check if req body is valid
-        if (!_.isArray(req.body) ||
-            !req.body.length) {
-            throw new Error('Invalid subscription request body.');
+    if (!_.isArray(req.body) || !req.body.length) {
+        return next(new restify.BadRequestError('Invalid subscription request body.'));
+    }
+    _.forEach(req.body, (s: {'correlationId': number;
```

Comment:
`_.every` is probably more appropriate for setting `isValid`. https://lodash.com/docs#every